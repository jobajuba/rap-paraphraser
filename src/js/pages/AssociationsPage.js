import React from "react"
import $ from "jquery"
import {Link} from "react-router-dom";
import {Header} from "../components/Header";
import {ResultField} from "../components/ResultField";

export class AssociationsPage extends React.Component {

    constructor(props) {
        super(props);
        this.clearField = this.clearField.bind(this);
        this.searchAssociates = this.searchAssociates.bind(this);
        this.state = { isLoading: false, data: [] };
    }

    clearField() {
        $("#enterWord").val("");
        this.state.data = [];
    }

    searchAssociates() {
        const word = $("#enterWord").val();
        this.setState({ isLoading: true });
        $.ajax("cgi-bin/associate.py", {
            data: { word },
            method: "GET",
            success: (data) => {
                this.setState({ data: JSON.parse(data), isLoading: false });
            }
        });
    }

    render() {

        const { data = [], isLoading } = this.state;
        let res = "";
        for (let i = 0; i < data.length; i++) {
            const dataItem = data[i];
            res += `${dataItem.word} (${dataItem.cos})\n`;
        }

        return (
            <React.Fragment>
                <Header/>
                <div className="size14 main-div">
                    <div id="div1">
                        <p>Введите слово:</p>
                        <form>
                            <input type="text" name="word" id="enterWord" defaultValue="" className="size14"/>
                            <br/>
                            <div className="button-bar">
                                <button type="button" className="size14" onClick={this.searchAssociates}>Поиск</button>
                                <button type="button" className="size14" onClick={this.clearField}>Очистить</button>
                            </div>
                        </form>
                    </div>
                    <div id="div2">
                        <p>Слова ассоциаты:</p>
                        <ResultField text = {res} isLoading = {isLoading} baseClass = "ws-pre" />
                        <div>
                            <Link to="/">Перейти на стартовую страницу</Link><br/>
                            <Link to="/paraphraser">Перейти к перефразировке текста</Link>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        )
    }
}