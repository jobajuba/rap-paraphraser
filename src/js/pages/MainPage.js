import React from "react"
import { Link } from "react-router-dom"
import { Header } from "../components/Header"

export const MainPage = () => {
    return (
        <>
            <Header/>
            <p className="center">На нашем сайте Вы можете:</p>
            <p className="center">
                <Link to="/paraphraser">перефразировать текст</Link><br/>
                <Link to="/associations">подобрать слова-ассоциаты</Link>
            </p>
            { /*<p className="center">а так же просмотреть примеры:</p>*/ }
        </>
    )
}
