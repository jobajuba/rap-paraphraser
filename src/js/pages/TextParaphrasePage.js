import React from "react"
import $ from "jquery"
import {Link} from "react-router-dom";
import {Header} from "../components/Header";
import {getTextWithReplaces} from "../helpers/replaceText";

export class TextParaphrasePage extends React.Component {

    constructor(props) {
        super(props);
        this.clearField = this.clearField.bind(this);
        this.paraphraseText = this.paraphraseText.bind(this);
        this.state = { isLoading: false, data: [] };
    }

    clearField() {
        $("#enterText").val("");
        $("#result").text("");
    }
    
    paraphraseText() {
        const text = $("#enterText").val();
        this.setState({ isLoading: true });
        $.ajax("cgi-bin/rephrase.py", {
            data: { text },
            method: "POST",
            success: (data) => {
                this.setState({ data: getTextWithReplaces(text, JSON.parse(data)), isLoading: false });
            }
        });
    }

    render() {
        const { data } = this.state;
        return (
            <React.Fragment>
                <Header/>
                <div className="size14 main-div">
                    <div id="div1">
                        <p>Введите текст:</p>
                        <form>
                            <textarea name="text" id="enterText" defaultValue="" className="size14"></textarea>
                            <br/>
                            <div className="button-bar">
                                <button type="button" className="size14" onClick={this.paraphraseText}>Отправить!</button>
                                <button type="button" className="size14" onClick={this.clearField}>Очистить</button>
                            </div>
                        </form>
                    </div>
                    <div id="div2">
                        <p>Измененный текст:</p>
                        <div id="result" className="ws-pre-wrap">{data}</div>
                        <div>
                            <Link to="/">Перейти на стартовую страницу</Link><br/>
                            <Link to="/associations">Перейти к поиску ассоциатов</Link>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        )
    }
}
