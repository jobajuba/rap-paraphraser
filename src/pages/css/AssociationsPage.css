.main-div { display: flex; font-size: 18px; }

.main-div > div { width: 50% }
.main-div > div:nth-of-type(1) { padding-right: 10px; }
.main-div > div:nth-of-type(2) { padding-left: 10px; }

.button-bar {
    text-align: center;
    width: 100%;
    margin-top: 10px;
}

.button-bar button { margin: 5px 5px; }

button, input { font-size: 18px; }

.enter-word { width: 100%; }

.ws-pre { white-space: pre; }
