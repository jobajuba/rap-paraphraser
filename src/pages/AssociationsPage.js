import React from 'react';
import {Link} from 'react-router-dom';
import {Header} from '../components/Header';
import {ResultField} from '../components/ResultField';
import * as api from '../api';
import './css/AssociationsPage.css';

export class AssociationsPage extends React.Component {
    constructor(props) {
        super(props);
        this.clearField = this.clearField.bind(this);
        this.searchAssociates = this.searchAssociates.bind(this);
        this.state = {isLoading: false, data: []};
    }

    clearField() {
        document.querySelector('.enter-word').value = '';
        this.setState({data: []});
    }

    searchAssociates() {
        const word = document.querySelector('.enter-word').value;
        this.setState({isLoading: true});
        api.searchAssociates(word)
            .then(data => this.setState({data: data, isLoading: false}));
    }

    render() {
        const {data = [], isLoading} = this.state;
        let res = '';
        for (let i = 0; i < data.length; i++) {
            const dataItem = data[i];
            res += `${dataItem.word} (${dataItem.cos})\n`;
        }

        return (
            <>
                <Header />
                <div className="main-div">
                    <div>
                        <p>Введите слово:</p>
                        <form>
                            <input type="text" name="word" className="enter-word" />
                            <br />
                            <div className="button-bar">
                                <button type="button" onClick={this.searchAssociates}>Поиск</button>
                                <button type="button" onClick={this.clearField}>Очистить</button>
                            </div>
                        </form>
                    </div>
                    <div>
                        <p>Слова ассоциаты:</p>
                        <ResultField text={res} isLoading={isLoading} baseClass="ws-pre" />
                        <div>
                            <Link to="/">Перейти на стартовую страницу</Link><br />
                            <Link to="/paraphraser">Перейти к перефразировке текста</Link>
                        </div>
                    </div>
                </div>
            </>
        );
    }
}
