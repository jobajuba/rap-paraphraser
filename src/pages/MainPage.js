import React from 'react';
import {Link} from 'react-router-dom';
import {Header} from '../components/Header';
import './css/MainPage.css';

export const MainPage = () => {
    return (
        <div className="main-page">
            <Header />
            <p>На нашем сайте Вы можете:</p>
            <p>
                <Link to="/paraphraser">перефразировать текст</Link><br />
                <Link to="/associations">подобрать слова-ассоциаты</Link>
            </p>
        </div>
    );
};
