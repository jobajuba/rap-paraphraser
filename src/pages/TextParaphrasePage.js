import React from 'react';
import {Link} from 'react-router-dom';
import {Header} from '../components/Header';
import {getTextWithReplaces} from '../helpers/replaceText';
import {ResultField} from '../components/ResultField';
import * as api from '../api';
import './css/TextParaphrasePage.css';

export class TextParaphrasePage extends React.Component {

    constructor(props) {
        super(props);
        this.clearField = this.clearField.bind(this);
        this.paraphraseText = this.paraphraseText.bind(this);
        this.state = {isLoading: false, data: ''};
    }

    clearField() {
        document.querySelector('.enter-text').value = '';
        this.setState({data: ''});

    }

    paraphraseText() {
        const text = document.querySelector('.enter-text').value;
        this.setState({isLoading: true});
        api.paraphraseText(text)
            .then(data => this.setState({
                data: getTextWithReplaces(text, data),
                isLoading: false,
            }));
    }

    render() {
        const {data = '', isLoading} = this.state;
        return (
            <>
                <Header />
                <div className="main-div">
                    <div>
                        <p>Введите текст:</p>
                        <form>
                            <textarea name="text" className="enter-text"/>
                            <br/>
                            <div className="button-bar">
                                <button type="button" onClick={this.paraphraseText}>
                                    Отправить!
                                </button>
                                <button type="button" onClick={this.clearField}>Очистить</button>
                            </div>
                        </form>
                    </div>
                    <div>
                        <p>Измененный текст:</p>
                        <ResultField text={data} isLoading={isLoading} baseClass="ws-pre-wrap"/>
                        <div>
                            <Link to="/">Перейти на стартовую страницу</Link><br/>
                            <Link to="/associations">Перейти к поиску ассоциатов</Link>
                        </div>
                    </div>
                </div>
            </>
        );
    }
}
