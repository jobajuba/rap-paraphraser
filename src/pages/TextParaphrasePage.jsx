import React, {useRef, useState} from 'react';
import {Link} from 'react-router-dom';
import {Header} from '../components/Header';
import {getTextWithReplaces} from '../helpers/replaceText';
import {ResultField} from '../components/ResultField';
import * as api from '../api';
import './css/TextParaphrasePage.css';

export const TextParaphrasePage = () => {
    const [isLoading, setLoading] = useState(false);
    const [data, setData] = useState('');
    const searchFieldRef = useRef(null);

    const clearField = () => {
        searchFieldRef.current.value = '';
        setData('');
    };

    const paraphraseText = async () => {
        const text = searchFieldRef.current.value;
        setLoading(true);

        const result = await api.paraphraseText(text);
        setData(getTextWithReplaces(text, result));
        setLoading(false);
    };

    return (
        <>
            <Header />
            <div className="main-div">
                <div>
                    <p>Введите текст:</p>
                    <form>
                        <textarea name="text" className="enter-text" ref={searchFieldRef} />
                        <br/>
                        <div className="button-bar">
                            <button type="button" onClick={paraphraseText}>
                                Отправить!
                            </button>
                            <button type="button" onClick={clearField}>Очистить</button>
                        </div>
                    </form>
                </div>
                <div>
                    <p>Измененный текст:</p>
                    <ResultField text={data} isLoading={isLoading} baseClass="ws-pre-wrap"/>
                    <div>
                        <Link to="/">Перейти на стартовую страницу</Link><br/>
                        <Link to="/associations">Перейти к поиску ассоциатов</Link>
                    </div>
                </div>
            </div>
        </>
    );
};
