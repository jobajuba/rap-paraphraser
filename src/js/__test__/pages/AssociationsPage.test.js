import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import {afterEach, beforeEach, describe, expect, test} from '@jest/globals';
import {BrowserRouter} from 'react-router-dom';
import {AssociationsPage} from '../../pages/AssociationsPage';

Enzyme.configure({adapter: new Adapter()});

describe('Test of AssociationsPage component', () => {

    beforeEach(() => {
        const div = document.createElement('div');
        div.setAttribute('id', 'testContainer');
        document.body.appendChild(div);
    });

    afterEach(() => {
        const div = document.querySelector('#testContainer');
        if (div) document.body.removeChild(div);
    });

    test('AssociationsPage component render', () => {
        const component = renderer.create(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Initial state is correct', () => {
        const component = mount(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>
        );
        expect(component.find('AssociationsPage').state()).toEqual({
            isLoading: false,
            data: [],
        });
    });

    test('AssociationsPage component render with data', () => {
        const component = mount(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>
        );

        const data = [
            {word: 'word1', cos: '0.1'},
            {word: 'word2', cos: '0.2'},
        ];
        const resultText = 'word1 (0.1)\nword2 (0.2)\n';

        component.find('AssociationsPage').setState({data});

        expect(component.find('ResultField').props().text).toEqual(resultText);
    });

    test('AssociationsPage component render with undefined data', () => {
        const component = mount(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>
        );

        component.find('AssociationsPage').setState({data: undefined});

        expect(component.find('ResultField').props().text).toEqual('');
    });

    test('clearField function is working', () => {
        const component = mount(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>,
            {attachTo: document.querySelector('#testContainer')}
        );

        const data = [
            {word: 'word1', cos: '0.1'},
            {word: 'word2', cos: '0.2'},
        ];
        component.find('AssociationsPage').setState({data});
        component.find('.button-bar button').at(1).simulate('click');

        expect(component.find('AssociationsPage').state().data).toHaveLength(0);
    });

    test ('searchAssociates function is working', () => {
        const component = mount(
            <BrowserRouter>
                <AssociationsPage/>
            </BrowserRouter>,
            {attachTo: document.querySelector('#testContainer')}
        );

        document.querySelector('#enterWord').value = 'testWord';

        let getURL = '', getConfig = {};

        const testGetPromise = new Promise((resolve) => {
            return resolve({response: {data: ['testResult']}});
        });

        // Заменяем axios.get для проверки корректности параметров
        axios.get = (url, config) => {
            getURL = url;
            getConfig = config;
            return testGetPromise;
        };

        component.find('.button-bar button').at(0).simulate('click');

        expect(getURL).toEqual('cgi-bin/associate.py');
        expect(getConfig).toEqual({params: {word: 'testWord'}});
    });

});
