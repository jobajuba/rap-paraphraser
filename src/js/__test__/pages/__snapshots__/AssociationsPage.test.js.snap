// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of AssociationsPage component AssociationsPage component render 1`] = `
Array [
  <h1>
    Rap-Paraphraser
  </h1>,
  <p
    id="p1"
  >
    Добро пожаловать на сайт для перефразирования текста любого типа (в том числе и одного слова) в семантическом поле русского репа с добавлением реп-сленга. Алгоритм перефразирования основан на использовании нейронной сети Word2Vec, обученной на текстах более, чем 9500 песен русских исполнителей, которая находит близкие по семантике слова в контексте русского репа.
  </p>,
  <div
    className="size14 main-div"
  >
    <div
      id="div1"
    >
      <p>
        Введите слово:
      </p>
      <form>
        <input
          className="size14"
          defaultValue=""
          id="enterWord"
          name="word"
          type="text"
        />
        <br />
        <div
          className="button-bar"
        >
          <button
            className="size14"
            onClick={[Function]}
            type="button"
          >
            Поиск
          </button>
          <button
            className="size14"
            onClick={[Function]}
            type="button"
          >
            Очистить
          </button>
        </div>
      </form>
    </div>
    <div
      id="div2"
    >
      <p>
        Слова ассоциаты:
      </p>
      <div
        className="ws-pre"
        id="result"
      >
        
      </div>
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          Перейти на стартовую страницу
        </a>
        <br />
        <a
          href="/paraphraser"
          onClick={[Function]}
        >
          Перейти к перефразировке текста
        </a>
      </div>
    </div>
  </div>,
]
`;
